### PR TITLE
detect-detection: Convert unittests to new FAIL/PASS API

### PR DESCRIPTION
**Pull request by hsadia538**
- [X] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [X] I have signed the Open Information Security Foundation contribution agreement at https://suricata.io/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Previous PR: https://github.com/OISF/suricata/pull/8115

Link to [redmine](https://redmine.openinfosecfoundation.org/issues/4019) ticket:

Describe new changes:

- Remove extra files

### CODE REVIEW

**Review by jufajardini — APPROVED** [review]
Judging from the previous PR, the code changes look good to me. 
I honestly don't understand why this complaint (https://github.com/OISF/suricata/actions/runs/3346628230/jobs/5543704496#step:12:537) if the vars are being used just a few lines down ... 🤔

**Comment by hsadia538** [comment]
> Judging from the previous PR, the code changes look good to me. I honestly don't understand why this complaint (https://github.com/OISF/suricata/actions/runs/3346628230/jobs/5543704496#step:12:537) if the vars are being used just a few lines down ... 🤔

Got it. They are not being used anymore in the updated version. removing these now.

File: src/detect-detection-filter.c [hunk 1]
```diff
@@ -611,24 +579,16 @@ static int DetectDetectionFilterTestSig3(void)
     drops += ((PacketTestAction(p, ACTION_DROP)) ? 1 : 0);
     p->action = 0;
 
-    if (alerts == 3 && drops == 3)
```
Comment (victorjulien) [inline src/detect-detection-filter.c:614]: what is there to replace these checks?

I think what needs to be done is:
1. remove the alerts and drops counters
2. use FAIL* macros for all `PacketAlertCheck` calls

File: src/detect-detection-filter.c [hunk 2]
```diff
@@ -611,24 +579,16 @@ static int DetectDetectionFilterTestSig3(void)
...
-        result = 1;
-    else {
-        if (alerts != 3)
-            printf("alerts: %d != 3: ", alerts);
-        if (drops != 3)
-            printf("drops: %d != 3: ", drops);
-    }
-
     SigGroupCleanup(de_ctx);
```
Comment (victorjulien) [inline src/detect-detection-filter.c:582]: these can be removed

**Review by victorjulien — CHANGES_REQUESTED** [review]
see inline comments

File: src/detect-detection-filter.c (hunk 1 above)
Comment (hsadia538) [reply src/detect-detection-filter.c:614]: One thing I am confused about is:
According to the above the test passes if `if (alerts == 3 && drops == 3)`
If I remove counters with FAIl/PASS macros then I will be failing or passing on all of these rather then a specific alert or drop count. Is that okay?
The alert and drop variables are getting incremented(or not) 7 times in the above lines.

**Comment by hsadia538** [comment]
Followed by #8262

**Pull request closed without merging**